arrows/ffmpeg/ffmpeg_init.cxx

```diff
@@ -35,22 +35,70 @@
 
 #include "ffmpeg_init.h"
 
+
+#include <algorithm>
+#include <cctype>
 #include <mutex>
 
 extern "C" {
 #include <libavformat/avformat.h>
 }
 
+#include <vital/logger/logger.h>
+
 std::mutex ffmpeg_init_mutex;
 
-//--------------------------------------------------------------------------------
+static auto ffmpeg_logger = kwiver::vital::get_logger( "arrows.ffmpeg" );
+
+//-----------------------------------------------------------------------------
+// trim whitespace from end of string (in place)
+static inline void rtrim(std::string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(),
+                         [](int ch) { return !std::isspace(ch);}
+                         ).base(), s.end());
+}
+
+//-----------------------------------------------------------------------------
+void
+ffmpeg_kwiver_log_callback(void* ptr, int level, const char* fmt, va_list vl)
+{
+  static int print_prefix = 0;
+  char line[1024];
+  av_log_format_line(ptr, level, fmt, vl, line, sizeof(line), &print_prefix);
+  std::string msg(line);
+  rtrim(msg);
+  switch(level)
+  {
+    case AV_LOG_PANIC:
+    case AV_LOG_FATAL:
+      LOG_ERROR(ffmpeg_logger, msg);
+      break;
+    case AV_LOG_ERROR:
```

Comment:
Is it intended that AV_ERRORs get logged as warnings?

Reply:
Yes,  FFMPEG has many more logging levels, as you can see.  I think "FATAL" is more inline with what we call an "ERROR".  I get hundreds of "ERROR" messages from FFMPEG on most videos because there is always some bad data somewhere, but there are recoverable, so I think KWIVER should treat them as warnings.

Reply:
Makes sense